x-pack/legacy/plugins/siem/server/lib/detection_engine/routes/rules/create_rules_bulk_route.test.ts

```diff
@@ -124,4 +124,18 @@ describe('create_rules_bulk', () => {
       expect(statusCode).toBe(400);
     });
   });
+
+  test('returns 409 if duplicate rule_ids found in request payload', async () => {
+    alertsClient.find.mockResolvedValue(getFindResult());
+    alertsClient.get.mockResolvedValue(getResult());
+    actionsClient.create.mockResolvedValue(createActionResult());
+    alertsClient.create.mockResolvedValue(getResult());
+    const request: ServerInjectOptions = {
+      method: 'POST',
+      url: `${DETECTION_ENGINE_RULES_URL}/_bulk_create`,
+      payload: [typicalPayload(), typicalPayload()],
+    };
+    const { payload } = await server.inject(request);
+    expect(JSON.parse(payload).error.status_code).toBe(409);
```

Comment:
👍 for all these tests.

We will have to be careful about what we determine is a 409 vs. a 200 with regards to bulk operations. At the moment this is slightly different behavior from the way the bulk works at the moment where if it sees an array of elements it tries to create each one and returns a 200 on the REST return.

The caller of the bulk create even if they get a 200 has to check the array elements to see if anything inside of it has been conflicted. So the bulk operations can actually return that only 1 item of the creates caused a conflict while the other elements were all success.

This mimics the saved objects bulk behaviors. I would suggest we do that here where if we see that there is a conflict we add it to the return array but try to create the other elements and still return a 200, but the caller of the API has to look at the return array to determine if and how many errors are present.

Reply:
I agree. I think this will make for a better user experience overall. I'll update the code. Thanks!